src/main/java/org/folio/list/services/export/CsvCreator.java

```diff
@@ -41,32 +42,54 @@ public class CsvCreator {
   private final ListExportProperties exportProperties;
   private final QueryClient queryClient;
   private final EntityTypeClient entityTypeClient;
+  private final FolioS3Client folioS3Client;
 
   @SneakyThrows
-  public ExportLocalStorage createCSV(ExportDetails exportDetails) {
+  public ExportLocalStorage createAndUploadCSV(ExportDetails exportDetails, String destinationFileName, String uploadId, List<String> partETags) {
     var localStorage = new ExportLocalStorage(exportDetails.getExportId());
     ListEntity list = exportDetails.getList();
     var idsProvider = new ListIdsProvider(contentsRepository, list);
     EntityType entityType = entityTypeClient.getEntityType(list.getEntityTypeId());
 
-    try (var localStorageOutputStream = localStorage.outputStream()) {
```

Comment:
Will removing this try-with-resources call cause any issues? We're not calling `close()` here, so I worry that removing the `try` could cause a regression

Reply:
If we remove the try-call, we need to call the "close" method directly. So let's leave it as it is, and then we can refactor it a little bit later